src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -1193,7 +1193,11 @@ def __call__(
                     noise_pred = rescale_noise_cfg(noise_pred, noise_pred_text, guidance_rescale=self.guidance_rescale)
 
                 # compute the previous noisy sample x_t -> x_t-1
+                old_dtype = latents.dtype
                 latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs, return_dict=False)[0]
+                if latents.dtype != old_dtype:
```

Comment:
Should this be guarded with `torch.backends.mps.is_available()` as well? It really seems to be happening only when `mps` is picked up no?

Reply:
(nit: I'd maybe call it `latents_dtype`; `old` sounds almost like it'd be ok if it changes).

Reply:
> Should this be guarded with `torch.backends.mps.is_available()`

In my opinion, it's ok the way it is as the comment already mentions `mps`, but no strong opinion.

Reply:
@sayakpaul @pcuenca it's about the outcome. if any other accelerator behaves in this way, would you rather it crash out of the box so that an issue is filed, or should we just invisibly fix it when we find it?

Reply:
How about throwing a warning so that the users are at least aware of it?

Reply:
or raise a specific error asking them to file a report?

Reply:
i modified the code so that it only executes on mps, as i figured it would be better to have full visibility into any other platform's dtype issues after further consideration.

the current error however, is pretty vague. it says two types are broadcast incompatible, which isn't very helpful to a new user

Reply:
Yeah raising an error sounds like a better idea. @pcuenca WDYT?